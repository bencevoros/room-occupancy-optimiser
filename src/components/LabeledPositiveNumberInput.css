label {
  display: flex;
  flex-direction: column;
}

label > input {
  padding: .3rem;
}
