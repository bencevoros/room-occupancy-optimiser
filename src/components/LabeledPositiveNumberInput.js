import './LabeledPositiveNumberInput.css';

function LabeledPositiveNumberInput({ id, label, onChange, value }) {
  return (
    <label htmlFor={id} data-testid={id}>
      {label}
      <input
        id={id}
        type="number"
        min="0"
        onChange={onChange}
        value={value}
      />
    </label>
  );
}

export default LabeledPositiveNumberInput;
