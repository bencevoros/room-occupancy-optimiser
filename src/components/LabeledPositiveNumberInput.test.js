import { render, screen } from '@testing-library/react';
import LabeledPositiveNumberInput from './LabeledPositiveNumberInput';

const label = 'Label';
const id = 'id';
const onChange = () => {};
const value = 0;

describe('LabeledPositiveNumberInput', () => {
  describe('calls setRoomOccupations', () => {
    test('when component renders', () => {
      render(
        <LabeledPositiveNumberInput id={id} label={label} onChange={onChange} value={value} />
      );
      expect(screen.queryByTestId(id)).toMatchSnapshot();
    });
  });
});
