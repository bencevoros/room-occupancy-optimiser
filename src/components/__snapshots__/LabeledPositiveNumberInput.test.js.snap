// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabeledPositiveNumberInput calls setRoomOccupations when component renders 1`] = `
<label
  data-testid="id"
  for="id"
>
  Label
  <input
    id="id"
    min="0"
    type="number"
    value="0"
  />
</label>
`;
